src/editors/activeConnectionCodeLensProvider.ts

```diff
@@ -1,10 +1,9 @@
 import * as vscode from 'vscode';
 
-export default class ActiveDBCodeLensProvider implements vscode.CodeLensProvider {
+export default class ActiveConnectionCodeLensProvider implements vscode.CodeLensProvider {
   private _codeLenses: vscode.CodeLens[] = [];
   private _connectionController: any;
-  private _runtime: any;
-  private _activeDB?: string;
+  private _activeConnectionName?: string;
```

Comment:
If a playground is opened or created before a connection is established, would this be `undefined`? I think another way we could do this which might allow us not to keep the connection name here would be to call `this._connectionController.getActiveConnectionName()` in the method `resolveCodeLens` for the connection name, and then we can handle the case where there is no active connection there. It also might allow us not to have the event listener `ACTIVE_CONNECTION_CHANGED` altogether.

Reply:
If a playground is opened or created before a connection is established, it will be no code lenses at all. So we don't want to show an active connection name when there is no connection. In case if we get rid of ACTIVE_CONNECTION_CHANGED listener we won't be able to update code lenses with active connection state if we didn't change a state of a playground. I mean we need to type something to invoke code lenses update. And I wanted to change the name in real-time.  Pls, try to play with a playground and let me think how does it feel like it works now.

Reply:
Ah gotcha. It looks like `ACTIVE_CONNECTION_CHANGED` is definitely a useful event to have. I think we can avoid storing the `_activeConnectionName` on the code lens though if we get the name from `this._connectionController.getActiveConnectionName()` in the `resolveCodeLens` method, since the code lens is resolved everytime `this._onDidChangeCodeLenses.fire()` is called. I'd like to avoid duplicating the storage of the connection name if possible, tried it on local and it seemed to do the trick.